framework/src/schema/application_config_schema.ts

```diff
@@ -338,22 +338,31 @@ export const applicationConfigSchema = {
 		},
 		rpc: {
 			type: 'object',
+			required: ['modes'],
 			properties: {
-				enable: {
-					type: 'boolean',
-				},
-				mode: {
-					type: 'string',
-					enum: ['ipc', 'ws'],
+				modes: {
+					type: 'array',
+					items: { type: 'string', enum: ['ipc', 'ws'] },
+					uniqueItems: true,
 				},
-				port: {
-					type: 'number',
-					minimum: 1024,
-					maximum: 65535,
+				ipc: {
+					type: 'object',
+					required: ['path'],
+					properties: {
+						path: { type: 'string' },
+					},
 				},
-				host: {
-					type: 'string',
-					format: 'ip',
+				ws: {
+					type: 'object',
+					required: ['host', 'port', 'path'],
```

Comment:
if we have `default` values, should it be `required`?

Reply:
In terms of `schema` and validation I think it must be required. Our validator does not support `default` keyword, we are doing `mergeDeep` on application level, so having restrict schema definition is better.